contracts/0.8.25/vaults/OperatorGrid.sol

```diff
@@ -0,0 +1,595 @@
+        if ($.tiers[_tierId].operator == address(0)) revert TierNotExists();
+
+        uint128 tierId = uint128(_tierId);
+
+        VaultTier storage vaultTier = $.vaultTier[_vault];
+        if (vaultTier.currentTierId == tierId) revert TierAlreadySet();
+        if (vaultTier.requestedTierId == tierId) revert TierAlreadyRequested();
+
+        vaultTier.requestedTierId = tierId;
+
+        address nodeOperator = IStakingVault(_vault).nodeOperator();
+        $.pendingRequests[nodeOperator].add(_vault); //returns true if the vault was not in the set
+
+        emit TierChangeRequested(_vault, vaultTier.currentTierId, _tierId);
+    }
+
+    /// @notice Confirm tier change request
+    /// @param _vault address of the vault
+    /// @param _tierIdToConfirm id of the tier to confirm
+    ///
+    /*
+
+    Legend:
+    V = Vault1.mintedShares
+
+    Scheme1 - transfer Vault from default tier to Tier2
+
+                                         ┌────────────────────────────────┐
+                                         │           Group 1              │
+                                         │                                │
+    ┌────────────────────┐               │  ┌───────────┐  ┌───────────┐  │
+    │  Tier 1 (default)  │   confirm     │  │ Tier 2    │  │ Tier 3    │  │
+    │  minted: -V        │    ─────▶     │  │ minted:+V │  │           │  │
+    └────────────────────┘               │  └───────────┘  └───────────┘  │
+                                         │                                │
+                                         │   Group1.mintedShares: +V      │
+                                         └────────────────────────────────┘
+
+    After confirmation:
+    - Tier 1.mintedShares   = -V
+    - Tier 2.mintedShares   = +V
+    - Group1.mintedShares   = +V
+
+    --------------------------------------------------------------------------
+    Scheme2 - transfer Vault from Tier2 to Tier3, no need to change group minted shares
+
+    ┌────────────────────────────────┐     ┌────────────────────────────────┐
+    │           Group 1              │     │           Group 2              │
+    │                                │     │                                │
+    │  ┌───────────┐  ┌───────────┐  │     │  ┌───────────┐                 │
+    │  │ Tier 2    │  │ Tier 3    │  │     │  │ Tier 4    │                 │
+    │  │ minted:-V │  │ minted:+V │  │     │  │           │                 │
+    │  └───────────┘  └───────────┘  │     │  └───────────┘                 │
+    │  operator1                     │     │  operator2                     │
+    └────────────────────────────────┘     └────────────────────────────────┘
+
+    After confirmation:
+    - Tier 2.mintedShares   = -V
+    - Tier 3.mintedShares   = +V
+
+    NB: Cannot changes from Tier2 to Tier1, because Tier1 has no group. Reverts on `requestTierChange`
+    NB: Cannot changes from Tier2 to Tier4, because Tier4 has different operator.
```

Comment:
```suggestion
    NB: Cannot change from Tier2 to Tier1, because Tier1 has no group. Reverts on `requestTierChange`
    NB: Cannot change from Tier2 to Tier4, because Tier4 has different operator.
```

Reply:
fixed